### DISCUSSION TITLE
How to use KubeOps in .net6 WebApi?

### DISCUSSION THREAD

**Discussion by zdiomaz** [dotnet/dotnet-operator-sdk#456] [Q&A] [answered]
I've created web api project using .net6 with KubeOps 6.5.3. It is configured such a way:
Program.cs
```
var builder = WebApplication.CreateBuilder(args);
builder.Services.AddKubernetesOperator();

var app = builder.Build();
app.UseKubernetesOperator();
app.RunOperatorAsync(args);
```

but when application starts in debug it is fails with error
![image](https://user-images.githubusercontent.com/5476410/174839752-4b44317a-1c6d-4e86-8169-3a1ae2e28929.png)

I try to create another totally the same web api project but using .net5 with KubeOps 6.0.18. It is configured such a way:
Program.cs
```
Host
  .CreateDefaultBuilder(args)
  .ConfigureWebHostDefaults(webBuilder => webBuilder.UseStartup<Startup>())
  .Build()
  .RunOperatorAsync(args);
```
Startup.cs
```
public void ConfigureServices(IServiceCollection services)
{
	services.AddKubernetesOperator();
}

public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
{
	app.UseKubernetesOperator();
}
```
... and it works fine.

What I'm doing wrong?

**Comment by buehler** [comment, marked as answer]
Could you provide some logs? What are the errors that appear when starting the operator?

**Reply by zdiomaz** [reply]
Instead `app.RunOperatorAsync(args);` I wrote `await app.RunOperatorAsync(args);` and it starts normally.
But it is strange that for .net 5 it works without `await`

**Reply by buehler** [reply]
hmm... this is quite the strange behaviour.

**Reply by wizpresso-steve-cy-fan** [reply]
> Instead `app.RunOperatorAsync(args);` I wrote `await app.RunOperatorAsync(args);` and it starts normally. But it is strange that for .net 5 it works without `await`

can confirm
